### PR TITLE
Autocompletion minor fixes

### LINKED ISSUE #11: Autocompletion
TODO: tests and fixes if needed

### PR DESCRIPTION
**Pull request by djfedos**
Autocompletions using dictrie

### CODE REVIEW

**Comment by djfedos** [comment]
@lioryah, FYI:
[this commit](https://github.com/djfedos/djfedos-boostrap/pull/10/commits/5fcff4b3b6fafe5198acfbe3c204531542267386) has several tests in it.
[this one](https://github.com/djfedos/djfedos-boostrap/pull/10/commits/a9a9c1c9f38ed4695abc2ad1d6bd9c97cb73d52f) has only one test in it

**Comment by djfedos** [comment]
the autocompletion/suggestion engine project is moved to its own repository:
[git@github.com:djfedos/djfedos-search.git](git@github.com:djfedos/djfedos-search.git)

**Comment by djfedos** [comment]
the autocompletion/suggestion engine project is moved to its own [repository](https://github.com/djfedos/djfedos-search)

File: notebooks/dynamic_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+O(2^n) time complexity
+O(n) space complexity
+"""
+
+fib_brute(9)
+
+#%%
+# Fibonacci numbers, memoization
+
+def fib(n, memo={}):
```
Comment (ydmeta) [inline notebooks/dynamic_tutorial.py:23]: this is nice usage of memoisation,
checkout `lru_cache` decorator from functools
@djfedos

File: notebooks/dynamic_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+# # this additional function is needed to kickstart the memoized function
+# # with the empty memo each time it's been called
+# # else there is always garbage stuck in the memo
+# def best_sum(target, nums):
+#     return best_sum_memo(target, nums, {})
+
+# print(best_sum_memo(100, [1, 2, 5, 25]))
+print(best_sum_memo(8, [1, 4, 5]))
+
+# %%
```
Comment (ydmeta) [inline notebooks/dynamic_tutorial.py:287]: this is a good example notebook with samples

**Pull request merged**